core/src/test/java/com/predic8/membrane/core/openapi/serviceproxy/APIProxyKeyTest.java

```diff
@@ -0,0 +1,94 @@
+/* Copyright 2023 predic8 GmbH, www.predic8.com
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+package com.predic8.membrane.core.openapi.serviceproxy;
+
+import com.predic8.membrane.core.http.Request.Builder;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.net.URISyntaxException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import static com.predic8.membrane.core.openapi.serviceproxy.APIProxyKey.additionalBasePaths;
+import static com.predic8.membrane.util.TestUtil.assembleExchange;
+import static java.util.Arrays.asList;
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.params.provider.Arguments.of;
+
+class APIProxyKeyTest {
+
+    APIProxyKey k1;
+
+    @BeforeEach
+    void setup() {
+        k1 = new APIProxyKey("","", 80, null) {{
+            addBasePaths(new ArrayList<>(List.of("/bar")));
+        }};
+    }
+
+    @Test
+    void complexMatchExpressionTrueTest() throws URISyntaxException {
+        var key = new APIProxyKey("", "", 80, "true");
+        var exc = new Builder().get("").buildExchange();
+        assertTrue(key.complexMatch(exc));
+    }
+
+    @Test
+    void complexMatchExpressionFalseTest() throws URISyntaxException {
```

Comment:
test if a http header is set with a certain value
test if a query parameter like param.customer == 7